src/replace.rs

```diff
@@ -85,9 +91,19 @@ impl Data {
         // the whole chunk. As an optimization and without loss of generality we
         // don't add empty parts.
         let new_parts = {
+            let is_inserted = |state| {
+                if let &State::Inserted(..) = state {
+                    true
+                } else {
+                    false
+                }
+            };
```

Comment:
Nice to see the closure work! Looking at how it is used here, it might be more idiomatic to write this as a method on State, though